tracing: Change tracing test to use ctr

cmaf opened the pull request:
Use ctr instead of docker for tracing test since 2.0 has no docker
support.

Fixes #3024

Signed-off-by: Chelsea Mafrica <chelsea.e.mafrica@intel.com>

jodh-intel reviewed (approved):
Thanks @cmaf.

jodh-intel commented:
/test

devimc reviewed (approved):
thanks @cmaf

cmaf commented:
@GabyCT or @chavafg, it seems like the tracing test should be running since it is part of the Makefile UNION variable and the jenkins jobs call "make test"; do you know if this test is disabled elsewhere? I don't see the output I expect (anything referring to spans) in the jenkins-ci-ubuntu-18-04 output.

chavafg commented:
@cmaf, I think that it is not being triggered as none of the current jobs is falling into the [default case](https://github.com/kata-containers/tests/blob/2.0-dev/.ci/run.sh#L131). So you have two options:
1) Add the `make tracing` call to one of the existing CI Jobs, maybe under: `CRI_CONTAINERD_K8S`.
2) Create a new `CI_JOB` which could be the default one, where only tracing test will be execute (and maybe in the future we can add other small tests), but you'll need to add another job in Jenkins for this.

/cc @GabyCT  @jcvenegas @jodh-intel for opinions on this.

cmaf commented:
/test

cmaf commented:
/test-ubuntu

cmaf commented:
/test-ubuntu

cmaf commented:
/test-ubuntu

cmaf commented:
/test-ubuntu

cmaf commented:
/test-ubuntu

cmaf commented:
/test-ubuntu

cmaf commented:
@chavafg I added it to an existing CI job but it will fail until [#904](https://github.com/kata-containers/kata-containers/pull/904) is merged. We may want to consider adding it to a non-required check.

cmaf commented:
/test-ubuntu

cmaf commented:
/test-ubuntu

cmaf commented:
This PR will test and is waiting until https://github.com/kata-containers/kata-containers/pull/904 (which is a forward port of https://github.com/kata-containers/runtime/pull/2980) is merged.

I know a lot of code was added since your last review but could you please re-review when you get time? @jodh-intel @devimc

devimc commented on tracing/tracing-test.sh:
```diff
@@ -229,48 +290,113 @@ check_jaeger_status()
+
+  CI    - if set, save logs of all tests to ${TRACE_LOG_DIR}.
+  DEBUG - if set, enable tracing and do not cleanup after tests.
+
```
and `DEBUG_KEEP_JAEGER` .. ?

devimc reviewed (commented):
thanks @cmaf - I've left some questions and comments

cmaf commented:
thanks @devimc.

Before I make changes and have it diverge too much from the original, could you take a look at some of these @jodh-intel for https://github.com/kata-containers/tests/pull/2129?

jodh-intel commented:
@cmaf - I've updated #2129, so once that lands, you can pull in the changes into this PR.

fidencio commented:
I'm really sorry for the disruption but in order to do the branch name changes announced [here](http://lists.katacontainers.io/pipermail/kata-dev/2021-January/001695.html) I'm adding the "do-not-merge" and the "needs-branch-adjustment" labels to the PR.

I'll add a new comment once the branch changes are done and once it happens I'd like to ask you to resubmit the PR against the new `main` branch, possibly linking to this PR in case you find the conversations / reviews already made important to proceed.

fidencio commented:
Kata Containers repos are back to business.  Thanks a lot for your patience and I'm very happy to announce that there's no changes needed from your side as renaming the branches automagically reflected here.

If you notice something weird, some missing CI, please, do not hesitate on contacting us here or on #kata-dev on bit.ly/katacontainersslack.

Also, this is a good opportunity to have your PR rebased atop of `main`, to ensure your PR follow all the guidelines (https://github.com/kata-containers/community/blob/master/CONTRIBUTING.md#patch-format) enforced by this community, or even to re-evaluate whether the PR is currently needed.

Last but not least,  as the default branch has been renamed to `main`, if you have a local clone you'd need to update it and you do so by running:
```
git branch -m 2.0-dev main
git fetch origin
git branch -u origin/main main
```

Happy hacking!

cmaf commented:
Test without polling for status

/test

cmaf commented:
/test-ubuntu

cmaf commented:
/test-ubuntu

jodh-intel reviewed (approved):
Nice! Thanks @cmaf.

lgtm

cmaf commented:
/test

cmaf commented:
There were a few updates to this PR since it was last approved:
1. I removed polling for jaeger output since it no longer is necessary based on testing that I did
2. Added the test to CRI_CONTAINERD_K8S_MINIMAL where it doesn't seem to add much additional time to the test
3. Modified the number of spans to check for to 50 since we expect the ctr command to yield at least that many spans
4. Misc syntactic changes, including changing the way errors are parsed in in check_jaeger_status()
5. Credited @jodh-intel's work from initial PR https://github.com/kata-containers/tests/pull/2129

@devimc do you think you could please review one more time since you're an original reviewer? cc @chavafg 

/test

devimc commented on tracing/tracing-test.sh:
```diff
@@ -9,13 +9,23 @@ set -o nounset
 
-RUNTIME=${RUNTIME:-kata-runtime}
+RUNTIME="io.containerd.kata.v2"
+CONTAINER_IMAGE="docker.io/library/busybox:latest"
```
please pull busybox from quay.io `quay.io/prometheus/busybox:latest`

devimc commented on tracing/tracing-test.sh:
```diff
@@ -25,17 +35,48 @@ jaeger_docker_container_name="jaeger"
+
+		# The tests worked so remove the logs
+		if [ -n "$DEBUG" ]; then
+			echo "-n DEBUG"
```
is this needed?

devimc commented on tracing/tracing-test.sh:
```diff
@@ -63,214 +104,250 @@ stop_jaeger()
+	get_trace_summary "$status" > "$logdir/span-summary.txt"
 
-	rm -f "$logfile"
+	return
```
this is the end of the function, is this needed?

devimc reviewed (approved):
thanks @cmaf - I've left some comments

cmaf commented:
@devimc thanks! I updated all based on your comments

/test

cmaf commented:
Ready to be merged once **jenkins-metrics-ubuntu-20-04** passes
